src/Thoth.Json.Net/Converters.fs

```diff
@@ -0,0 +1,25 @@
+module Thoth.Json.Net.Converters
+
+open Newtonsoft.Json
+open Newtonsoft.Json.Linq
+
+type Converter (?isCamelCase : bool, ?extra: ExtraCoders) =
+    inherit JsonConverter()
+
+    override __.CanConvert(t) = true
+
+    override __.WriteJson(writer, value, _) = 
+        let t = value.GetType()
+        let encoder = Encode.Auto.generateEncoderCached(t, ?isCamelCase=isCamelCase, ?extra=extra)
+        (encoder value).WriteTo(writer)
+        writer.Flush()
+        
+    override __.ReadJson(reader, t, _, _) =
+        let decoder = Decode.Auto.generateDecoderCachedByType(t, ?isCamelCase=isCamelCase, ?extra=extra)
```

Comment:
I would use the version which use the generic type here too (same as above)

Reply:
Well, it is not like above. t is of System.Type (and not generic as the value)